### PR TITLE
test(es/minifier): Add a test for `??` operator

### PR DESCRIPTION
**Pull request by kdy1**
**Description:**

I'm still investigating.

**Related issue:**

 - https://github.com/vercel/next.js/issues/41992.

---

# Investigation

```rust
                        &JsMinifyOptions {
                            compress: BoolOrDataConfig::from_bool(false),
                            mangle: BoolOrDataConfig::from_bool(false),
                            ..self.opts.clone()
                        }
```

works, while

```rust
                        &JsMinifyOptions {
                            compress: TerserCompressorOptions {
                                defaults: false,
                                ..Default::default()
                            }
                            .into(),
                            mangle: BoolOrDataConfig::from_bool(false),
                            ..self.opts.clone()
                        },

```

does not

----

I found the cause.

image

is compiled as 

image

so we should preserve `??`

### CODE REVIEW

File: crates/swc_ecma_minifier/tests/fixture/next/41992/output.js [hunk 1]
```diff
@@ -0,0 +1,184 @@
...
+                        callbackVisible: !1,
+                        parametersVisible: !0
+                    };
+                }
+                render() {
+                    var e;
+                    let { onTryoutClick: t , parameters: r , allowTryItOut: a , tryItOutEnabled: n , specPath: l , fn: s , getComponent: o , getConfigs: i , specSelectors: u , specActions: d , pathMethod: g , oas3Actions: v , oas3Selectors: E , operation: b  } = this.props;
+                    const S = o("parameterRow"), _ = o("TryItOutButton"), w = o("contentType"), C = o("Callbacks", !0), x = o("RequestBody", !0), A = n && a, I = u.isOAS3(), R = b.get("requestBody"), N = (0, p.default)(e = (0, ft.default)((0, p.default)(r).call(r, (e, t)=>{
+                        const r = t.get("in");
+                        return e[r] ?? (e[r] = []), e[r].push(t), e;
```
Comment (kdy1) [inline crates/swc_ecma_minifier/tests/fixture/next/41992/output.js:44]: This was previously `e[r], e[r] = []`

**Review by kdy1 — COMMENTED** [review]
swc-bump:
 - swc_ecma_minifier

**Review by jridgewell — DISMISSED** [review]

**Comment by jridgewell** [comment]
Is it possible to add a reduced test case?

File: crates/swc_ecma_minifier/tests/fixture/next/41992/1/output.js [hunk 2]
```diff
@@ -0,0 +1,184 @@
...
+                    }))));
+                }
+            }
+            (0, W.default)(ht, "defaultProps", {
+                onTryoutClick: Function.prototype,
+                onCancelClick: Function.prototype,
+                tryItOutEnabled: !1,
+                allowTryItOut: !0,
+                onChangeKey: [],
+                specPath: []
```
Comment (kdy1) [inline crates/swc_ecma_minifier/tests/fixture/next/41992/1/output.js:180]: @jridgewell Do you think it's better to remove this file?

Comment (jridgewell) [reply crates/swc_ecma_minifier/tests/fixture/next/41992/1/output.js:180]: Yes. If this file changes in the future, there's very little chance I'll spot the previously buggy code again.

Comment (kdy1) [reply crates/swc_ecma_minifier/tests/fixture/next/41992/1/output.js:180]: Good idea, I'll do it for other patches too

**Comment by Austaras** [comment]
Has this already been fixed by #6272?

**Comment by kdy1** [comment]
@Austaras No, as we have removing logic in more places

**Comment by Austaras** [comment]
No, there's alreay a `may_short_circuit` in L668, and I've testes the input on main branch which generates identical output.

**Comment by Austaras** [comment]
The `ignore_return_value` is really too long to parse for human...

**Comment by kdy1** [comment]
Oh, if then it's possible that I was using a old base branch.

**Comment by kdy1** [comment]
You are right, this was fixed already and I was using an old base

**Pull request merged**